### DISCUSSION TITLE
Strategy against Supply Chain Attacks

### DISCUSSION THREAD

**Discussion by mfhepp** [keepassxreboot/keepassxc#8382] [Ideas]
Just wondering: What is the strategy against [Supply Chain Attacks](https://docs.microsoft.com/en-us/microsoft-365/security/intelligence/supply-chain-malware?view=o365-worldwide)? Given that KeePassXC is a super-attractive target, and the growing number of attempts to compromise common libraries (see e.g. [this one on PyPi](https://arstechnica.com/information-technology/2022/08/10-malicious-python-packages-exposed-in-latest-repository-attack/)), I am wondering how KeePassXC can be protected against this.

Please note that I am not so much concerned of malicious pull-requests against the actual KeePassXC sources (even though there are many possible tricks to obscure malicious additions, e.g. as shown [here](https://github.com/mortenson/pr-sneaking) or potential attack vectors in typical [Github-based CI-workflows](https://nathandavison.com/blog/github-actions-and-the-threat-of-malicious-pull-requests).

My main concern are targeted attempts to compromise upstream repositories.

Is this on the radar? If so, what are mitigation strategies?

**Edit:**

Most likely targets for such Supply Chain attacks would be the documented [build dependencies like Qt5](https://github.com/keepassxreboot/keepassxc/blob/develop/INSTALL.md).

We will have to **consider malicious insertions there that specifically target KeePassXC,** i.e. they would remain inactive and hence most likely undetected in other usages of these libraries.

**Comment by droidmonkey** [comment]
Short answer: you can never fully protect yourself from this, if you don't own the FULL stack then there will always be a possible way to introduce an upstream attack. 

We limit our dependencies to the bare minimum possible. We also keep them up to date with every release. This would be classified as a risk reduction approach, which is the best we can do.
